saleor/graphql/notifications/mutations/external_notification_trigger.py

```diff
@@ -0,0 +1,142 @@
+from saleor.core.permissions import (
+    CheckoutPermissions,
+    OrderPermissions,
+    ProductPermissions,
+)
+from saleor.graphql.core.types.common import ExternalNotificationError
+from saleor.invoice.models import Invoice
+from saleor.order.models import Fulfillment, Order
+from saleor.product.models import Product, ProductVariant
+
+from ....core.permissions import AccountPermissions
+from ....graphql.utils import resolve_global_ids_to_primary_keys
+from ....webhook.payloads import (
+    generate_checkout_payload,
+    generate_customer_payload,
+    generate_fulfillment_lines_payload,
+    generate_invoice_payload,
+    generate_order_payload,
+    generate_product_payload,
+    generate_product_variant_payload,
+)
+from ...core.mutations import BaseMutation
+from ..inputs import ExternalNotificationTriggerInput
+
+
+class ExternalNotificationTrigger(BaseMutation):
+    PK__IN = "pk__in"
+    PAYLOAD_MAPPING = {
+        "Fulfillment": (
+            Fulfillment,
+            generate_fulfillment_lines_payload,
+            None,
+            PK__IN,
+            OrderPermissions.MANAGE_ORDERS,
+        ),
+        "ProductVariant": (
+            ProductVariant,
+            generate_product_variant_payload,
+            list,
+            PK__IN,
+            ProductPermissions.MANAGE_PRODUCTS,
+        ),
+        "Product": (
+            Product,
+            generate_product_payload,
+            None,
+            PK__IN,
+            ProductPermissions.MANAGE_PRODUCTS,
+        ),
+        "User": (
+            User,
+            generate_customer_payload,
+            None,
+            PK__IN,
+            AccountPermissions.MANAGE_USERS,
+        ),
+        "Checkout": (
+            Checkout,
+            generate_checkout_payload,
+            None,
+            "token__in",
+            CheckoutPermissions.MANAGE_CHECKOUTS,
+        ),
+        "Invoice": (
+            Invoice,
+            generate_invoice_payload,
+            None,
+            PK__IN,
+            OrderPermissions.MANAGE_ORDERS,
+        ),
+        "Order": (
+            Order,
+            generate_order_payload,
+            None,
+            PK__IN,
+            OrderPermissions.MANAGE_ORDERS,
+        ),
+    }
+
+    class Arguments:
+        input = ExternalNotificationTriggerInput(
+            required=True, description="Input for External Notification Trigger."
+        )
+        pluginId = graphene.String(description="The ID of notification plugin.")
+
+    class Meta:
+        description = "Send external notification to given customers."
+        error_type_class = ExternalNotificationError
+
+    @classmethod
+    def perform_mutation(cls, root, info, **data):
+        manager = info.context.plugins
+        ids, extra_payloads, external_event_type = cls.get_cleaned_data(data)
+        model_type, pks = resolve_global_ids_to_primary_keys(ids)
+        if payload_params := cls.PAYLOAD_MAPPING.get(model_type):
```

Comment:
We shouldn't mix in mutation logic with business logic.  Mutation responsibility is to validate received data, clean up data,  and call a proper action.
All things related to prepare notification payload, mapping, and calling `notify` should be extracted to a separate file. I would consider if we can extract it to saleor/core/notification file.

Reply:
I think we could leave it as it is. If there was more logic then yes, but in this case, this mutation has a very simple responsibility.